server/proxy/contexts.go

```diff
@@ -47,15 +47,65 @@ func (cp *contextsProxy) List(ctx context.Context, request *contextsv1.ListReque
 		return &contextsv1.ListResponse{}, err
 	}
 
+	return convertContexts(contexts, configFile.CurrentContext), nil
+}
+
+func convertContexts(contexts []*store.DockerContext, currentContext string) *contextsv1.ListResponse {
 	result := &contextsv1.ListResponse{}
 
 	for _, c := range contexts {
-		result.Contexts = append(result.Contexts, &contextsv1.Context{
+		endpointName := c.Type()
+		if c.Type() == store.DefaultContextType {
+			endpointName = "docker"
+		}
+		var endpoint interface{} = c.Endpoints[endpointName]
+
+		context := contextsv1.Context{
 			Name:        c.Name,
 			ContextType: c.Type(),
-			Current:     c.Name == configFile.CurrentContext,
-		})
+			Description: c.Metadata.Description,
+			Current:     c.Name == currentContext,
+		}
+		switch c.Type() {
+		case store.DefaultContextType:
+			context.Endpoint = getDockerEndpoint(endpoint)
+		case store.AciContextType:
+			context.Endpoint = getAciEndpoint(endpoint)
+		case store.EcsContextType:
+			context.Endpoint = getEcsEndpoint(endpoint)
```

Comment:
We should have a default here so that we handle all cases. Specifically as we have the local and example types as well.

Reply:
I'll add a default with no endpoint info for now. (and we'll add specific endpoint as we add backends). 
Local backend is not yet available in released CLI, I think we can handle this in a separate PR

Reply:
done